test/test_nn.py

```diff
@@ -5017,6 +5017,17 @@ def test_constant(self):
 
                 self.assertEqual(input_tensor, input_tensor.clone().fill_(val))
 
+    def test_ones_and_zeros(self):
+        for init_fn_, val in zip([init.ones_, init.zeros_], [1, 0]):
+            for as_variable in [True, False]:
+                for dims in [1, 2, 4]:
+                    input_tensor = self._create_random_nd_tensor(dims, size_min=1, size_max=5, as_variable=as_variable)
+                    init_fn_(input_tensor)
+                    if as_variable:
+                        input_tensor = input_tensor.data
```

Comment:
I really don’t get what’s going on with the as_variable part. Everything is a variable now, so you could remove the loop and this conditional